### PR TITLE
64 bit: fix test failures

### PR DESCRIPTION
**Pull request by fzi-haxel**
### Contribution description

- flashpage-test: Fixed print format and removed C preprocessor if statement.
   `sizeof(uintptr_t) == sizeof(uint32_t)` does not imply `PRIx32 == PRIxPTR`.
    While it worked for x86-gcc, the arm-gcc resolves `uintptr_t` to `unsigned int` and `uint32_t` to `unsigned long`.
- sys: Swapped `PRIuSIZE` format specifiers
- tests-uri_parser.c: Readded missing `unsigned` cast
- tests-fib.c: Reverting `size_t` to `unsigned` type change for `add_buf_size` due to `fib_get_next_hop(..., &add_buf_size, ...)` call. I was a bit to eager with this change

### Testing procedure
flashpage-test: BOARD=iotlab-m3 make -C tests/periph/flashpage
Others: Compiled all examples and tests for avr-rss2,  iotlab-m3, esp32-wroom-32, native and "native64"

### CODE REVIEW

File: sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c [hunk 1]
```diff
@@ -243,7 +243,7 @@ gnrc_pktsnip_t *gnrc_pktbuf_start_write(gnrc_pktsnip_t *pkt)
 #ifdef MODULE_OD
 static inline void _print_chunk(void *chunk, size_t size, int num)
 {
-    printf("=========== chunk %3" PRIuSIZE " (%-10p size: %4u) ===========\n", num, chunk,
+    printf("=========== chunk %3u (%-10p size: %4" PRIuSIZE ") ===========\n", num, chunk,
```
Comment (OlegHahm) [inline sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c]: Shouldn't that be `%3i`?

File: sys/net/gnrc/pktdump/gnrc_pktdump.c [hunk 2]
```diff
@@ -151,7 +151,7 @@ static void _dump(gnrc_pktsnip_t *pkt)
     gnrc_pktsnip_t *snip = pkt;
 
     while (snip != NULL) {
-        printf("~~ SNIP %2" PRIuSIZE " - size: %3u byte, type: ", snips,
+        printf("~~ SNIP %2u - size: %3" PRIuSIZE " byte, type: ", snips,
```
Comment (OlegHahm) [inline sys/net/gnrc/pktdump/gnrc_pktdump.c]: `%2i`

File: sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c (hunk 1 above)
Comment (fzi-haxel) [reply sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c]: Yes, you are absolutely correct.

File: sys/net/gnrc/pktdump/gnrc_pktdump.c (hunk 2 above)
Comment (fzi-haxel) [reply sys/net/gnrc/pktdump/gnrc_pktdump.c]: Same

**Comment by fzi-haxel** [comment]
Added changes a separate PR.

**Pull request closed without merging**